integrations/google-tag-manager/lib/index.js

```diff
@@ -53,7 +61,10 @@ GTM.prototype.initialize = function() {
  */
 
 GTM.prototype.loaded = function() {
-  return !!(window.dataLayer && Array.prototype.push !== window.dataLayer.push);
+  return !!(
+    window[this.options.dataLayer] &&
```

Comment:
Assume `loaded` runs after `initialize`? Otherwise `this.options.dataLayer` might be undefined?

Reply:
Maybe add a comment clarifying that... even if obvious to experienced 'integrations authors'.

Reply:
Hmmm my understanding is the `loaded` method is used exclusively in our test suite.